src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -4222,6 +4270,21 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
                 }
                 break;
             }
+            case (ASR::cast_kindType::RealToCharacter) : {
+                llvm::Value *arg = tmp;
+                tmp = lfortran_float_to_str(arg);
+                break;
+            }
+            case (ASR::cast_kindType::IntegerToCharacter) : {
+                llvm::Value *arg = tmp;
+                tmp = lfortran_int_to_str(arg);
+                break;
+            }
+            case (ASR::cast_kindType::LogicalToCharacter) : {
+                llvm::Value *arg =tmp;
+                tmp = lfortran_bool_to_str(arg);
```

Comment:
Don't call into C here. This can be done in LLVM itself.

```suggestion
                llvm::Value *cmp = builder->CreateICmpEQ(tmp, builder->getInt1(0));
                llvm::Value *zero_str = builder->CreateGlobalStringPtr("False");
                llvm::Value *one_str = builder->CreateGlobalStringPtr("True");
                tmp = builder->CreateSelect(cmp, zero_str, one_str);
```